### PR TITLE
Upgrade eslint and improve some rules

### LINKED ISSUE #18: Configure eslint more
https://github.com/yannickcr/eslint-plugin-react

### PR DESCRIPTION
**Pull request by grapegravity**
- require empty line at end of file (`eol-last`)
- disallow duplicate function arguments (`no-dupe-args`)
- disallow multiple empty lines (`no-multiple-empty-lines`)
- require variables to be declared separately (`one-var`)
- disallow spaces before function parens (`space-before-function-paren`)
- require display name in all react components (`react/display-name`)
- disallow using undefined varialbes in JSX (`react/jsx-no-undef`)
- require double quotes in JSX (`react/jsx-quotes`)
- make JSX treat `React` as a used variable (`react/jsx-uses-react`)
- treat variables used in JSX as used variables (`react/jsx-uses-vars`)
- discourage setting state in component mount and update (`react/no-did-mount-set-state` and `react/no-did-update-set-state`)
- discourage using unknown DOM properties (`react/no-unknown-property`)
- warn if React is missing in JSX files (`react/react-in-jsx-scope`)
- require tags with no children to self-close (`react/self-closing-comp`)

fixes #18 

I’ve tested this on `macropod-components` and it looks good. Let me know if anything should be reclassified/removed/made more strict!

### CODE REVIEW

**Comment by nelix** [comment]
https://github.com/babel/babel-eslint then we can remove all the silly feature language configurations and get propper not broken jsx and es6 class support

**Comment by grapegravity** [comment]
Added in a7a072d.

File: package.json [hunk 1]
```diff
@@ -35,5 +35,8 @@
     "macropod-build": "./bin/build.sh",
     "macropod-start-test": "./bin/start-test.sh"
   },
-  "homepage": "https://github.com/macropodhq/macropod-tools"
+  "homepage": "https://github.com/macropodhq/macropod-tools",
+  "devDependencies": {
+    "babel-eslint": "^2.0.2"
```
Comment (nelix) [inline package.json]: I think this needs to be a full dependency or else it won't work when required in other projects

Comment (grapegravity) [inline package.json]: Good point!

**Pull request merged**